src/argilla/client/feedback/dataset/base.py

```diff
@@ -298,19 +212,44 @@ def question_by_name(self, name: str) -> AllowedQuestionTypes:
             f" {', '.join(q.name for q in self._questions)}"
         )
 
-    @property
-    def records(self) -> List[FeedbackRecord]:
-        """Returns the all the records in the dataset."""
-        return self._records + self._new_records
+    def _validate_records(
```

Comment:
Not for this PR but, I think it make sense to validate records also based on dataset questions. Maybe records contain incompatible responses for configured dataset questions